scripts/setup-firebase-flavors.sh

```
@@ -0,0 +1,185 @@
+    # Ensure we're in the app directory
+    ensure_app_directory
+    
+    # Create directory structure
+    create_directory_structure
+    
+    # Check if specific flavor is provided
+    if [ $# -eq 1 ]; then
+        local requested_flavor=$1
+        
+        # Validate if requested flavor is in the list
+        if [[ " ${FLAVORS[*]} " =~ " ${requested_flavor} " ]]; then
```

Shellcheck (0.10.0)

[warning] Remove quotes from right-hand side of =~ to match as a regex rather than literally.

(SC2076)